src/internal/pachtmpl/pachtmpl.go

```diff
@@ -24,11 +26,18 @@ func ParseArgs(argStrs []string) (map[string]string, error) {
 }
 
 // RenderTemplate renders the template tmpl, using args and returns the result.
-func RenderTemplate(tmpl string, args map[string]string) (string, error) {
+func RenderTemplate(tmpl string, args *structpb.Struct) (string, error) {
 	vm := newVM(nil)
-	for key, value := range args {
-		vm.TLAVar(key, value)
+	for key, value := range args.Fields {
+		switch value.Kind.(type) {
+		case *structpb.Value_BoolValue:
+			boolNode := &ast.LiteralBoolean{Value: value.GetBoolValue()}
+			vm.TLANode(key, boolNode)
+		default:
+			vm.TLAVar(key, value.GetStringValue())
```

Comment:
We will need to add the other possible struct values here or figure out how to cast them to strings

```
    // Represents a null value.
    NullValue null_value = 1;
    // Represents a double value.
    double number_value = 2;
    // Represents a string value.
    string string_value = 3;
    // Represents a boolean value.
    bool bool_value = 4;
    // Represents a structured value.
    Struct struct_value = 5;
    // Represents a repeated `Value`.
    ListValue list_value = 6;
```